mlir/test/Dialect/Vector/vector-emulate-narrow-type.mlir

```diff
@@ -439,6 +439,10 @@ func.func @vector_store_i4(%arg0: vector<8xi4>, %arg1: index, %arg2: index) {
 
 // -----
 
+// FIXME: This example assumes that the store happens at a byte boundary, but
+// that's not guaranteed. Below is a counter-example with specific dimensions:
+//    vector.store %arg0, %0[0, 3] : memref<2x13xi4>, vector<8xi4>
```

Comment:
side note: I am trying to add a pass to "linearize" all memrefs: https://github.com/llvm/llvm-project/pull/136797

I think there is already a linearizer for vectors.

So in the future we only need to deal with 1-d memrefs.